packages/smooth_app/lib/background/background_task_hunger_games.dart

```diff
@@ -0,0 +1,164 @@
+        'processName': processName,
+        'uniqueId': uniqueId,
+        'barcode': barcode,
+        'languageCode': languageCode,
+        'user': user,
+        'country': country,
+        'stamp': stamp,
+        'insightId': insightId,
+        'insightAnnotation': insightAnnotation,
+      };
+
+  /// Returns the deserialized background task if possible, or null.
+  static AbstractBackgroundTask? fromJson(final Map<String, dynamic> map) {
+    try {
+      final AbstractBackgroundTask result =
+          BackgroundTaskHungerGames._fromJson(map);
+      if (result.processName == _PROCESS_NAME) {
+        return result;
+      }
+    } catch (e) {
+      //
+    }
+    return null;
+  }
+
+  /// Adds the background task about hunger games.
+  static Future<void> addTask({
+    required final String barcode,
+    required final String insightId,
+    required final InsightAnnotation insightAnnotation,
+    required final State<StatefulWidget> widget,
+  }) async {
+    final LocalDatabase localDatabase = widget.context.read<LocalDatabase>();
+    final String uniqueId = await _operationType.getNewKey(
+      localDatabase,
+      barcode,
+    );
+    final AbstractBackgroundTask task = _getNewTask(
+      barcode,
+      insightId,
+      insightAnnotation.value,
+      uniqueId,
+    );
+    await task.addToManager(localDatabase, widget: widget);
+  }
+
+  @override
+  String? getSnackBarMessage(final AppLocalizations appLocalizations) => null;
+
+  /// Returns a new background task about hunger games.
+  static BackgroundTaskHungerGames _getNewTask(
+    final String barcode,
+    final String insightId,
+    final int insightAnnotation,
+    final String uniqueId,
+  ) =>
+      BackgroundTaskHungerGames._(
+        processName: _PROCESS_NAME,
+        uniqueId: uniqueId,
+        barcode: barcode,
+        languageCode: ProductQuery.getLanguage().offTag,
+        user: jsonEncode(ProductQuery.getUser().toJson()),
+        country: ProductQuery.getCountry()!.offTag,
+        stamp: _getStamp(barcode, insightId),
+        insightId: insightId,
+        insightAnnotation: insightAnnotation,
+      );
+
+  static String _getStamp(final String barcode, final String insightId) =>
+      '$barcode;hungerGames;$insightId';
+
+  @override
+  Future<void> preExecute(final LocalDatabase localDatabase) async {}
+
+  @override
+  Future<void> postExecute(
+    final LocalDatabase localDatabase,
+    final bool success,
+  ) async {
+    await super.postExecute(localDatabase, success);
+    final RobotoffInsightHelper robotoffInsightHelper =
+        RobotoffInsightHelper(localDatabase);
+    await robotoffInsightHelper.cacheInsightAnnotationVoted(
+      barcode,
+      insightId,
+    );
+  }
+
+  /// Unselects the product image.
+  @override
+  Future<void> upload() async {
+    final InsightAnnotation? annotation =
+        _getInsightAnnotation(insightAnnotation);
+    if (annotation == null) {
```

Comment:
Wouldn't be a good idea to log this kind of error to measure the volume of potentials errors?

Reply:
It's very very unlikely to happen, as we previously populated the task field as `InsightAnnotation`.
Feel free to create an issue if really needed.